flink-runtime/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -323,6 +355,15 @@ public void validateRunsInMainThread() {
 		assert MainThreadValidatorUtil.isRunningInExpectedThread(currentMainThread.get());
 	}
 
+	/**
+	 * Rpc endpoint's state.
+	 */
+	public enum State {
```

Comment:
Do we have a context why State in `RpcEndpoint` designed to be `CREATED`, `STARTED` and `STOPPING`?

At least document as javadoc to express what the state exactly means and transitions could be help.

Reply:
The states basically reflect the lifecycle of the RPC Endpoint (as written in JIRA) and what one can do with it by calling the available methods: start/stop/close and getting callbacks: onStart and onStop.
Then the concrete endpoints can easier judge about the state in concrete API calls (also in JIRA).
I can add a jdoc comment about when the transitions happen.

Reply:
Thanks for your reply. 

From JIRA it says "we suggest to introduce its state:..." without why the exactly states.

It would be helpful if you write down the transitions. As @zentol's comment, you can find `STOPPING` is not yet well-defined.

Let's say we prefer reaching a consensus in JIRA first on which states we will introduce, accurate meaning of state, and when transition happens. Here is one way to implement it and maybe it'd better to discuss the requirement on JIRA first.

Reply:
It says `To better reflect the lifecycle of RpcEndpoint..`
I will write it down after we fully confirm them here. I expect a review from @tillrohrmann as we agreed offline.